runtime-modules/proposals/engine/src/lib.rs

```diff
@@ -898,16 +845,101 @@ type ApprovedProposal<T> = ApprovedProposalData<
     <T as Trait>::ProposalId,
     <T as system::Trait>::BlockNumber,
     MemberId<T>,
-    types::BalanceOf<T>,
-    <T as stake::Trait>::StakeId,
+    BalanceOf<T>,
     <T as system::Trait>::AccountId,
 >;
 
 // Simplification of the 'Proposal' type
 type ProposalOf<T> = Proposal<
     <T as system::Trait>::BlockNumber,
     MemberId<T>,
-    types::BalanceOf<T>,
-    <T as stake::Trait>::StakeId,
+    BalanceOf<T>,
     <T as system::Trait>::AccountId,
 >;
+
+pub struct StakingManager<T: Trait, LockId: Get<LockIdentifier>> {
+    trait_marker: PhantomData<T>,
+    lock_id_marker: PhantomData<LockId>,
+}
+
+impl<T: Trait, LockId: Get<LockIdentifier>> StakingHandler<T> for StakingManager<T, LockId> {
```

Comment:
This should not be in the engine itself, that makes the engine as a software module, tightly coupled to the balances and membership modules. It should be in the runtime, the same applies to the working group case.

Reply:
I agree with both. It is situated here until we integrate at least 2 modules.

Reply:
Ok.